xarray/backends/netCDF4_.py

```diff
@@ -485,21 +496,20 @@ def encode_variable(self, variable):
         return variable
 
     def prepare_variable(
-        self, name, variable, check_encoding=False, unlimited_dims=None
+        self, name, variable: Variable, check_encoding=False, unlimited_dims=None
     ):
         _ensure_no_forward_slash_in_name(name)
-
-        datatype = _get_datatype(
-            variable, self.format, raise_on_invalid_encoding=check_encoding
-        )
         attrs = variable.attrs.copy()
-
         fill_value = attrs.pop("_FillValue", None)
-
         encoding = _extract_nc4_variable_encoding(
```

Comment:
Move this below the following if-clause. There is some encoding-changes going on in `_get_datatype`. This will fix the `test_encoding_kwarg_vlen_string` errors.